### PR TITLE
DOC: Fix indexing docs to pass refguide

### PR DESCRIPTION
**Pull request by pdebuyl**
Related to #14970

This PR fixes the refguide_check.py for arrays.indexing.rst

### CODE REVIEW

File: doc/source/reference/arrays.indexing.rst [hunk 1]
```diff
@@ -265,7 +269,7 @@ understood with an example.
     one needs to select all elements *explicitly*. Using the method explained
     previously one could write:
 
-    >>> x = array([[ 0,  1,  2],
+    >>> x = np.array([[ 0,  1,  2],
     ...            [ 3,  4,  5],
     ...            [ 6,  7,  8],
     ...            [ 9, 10, 11]])
```
Comment (eric-wieser) [inline doc/source/reference/arrays.indexing.rst]:
```suggestion
    >>> x = np.array([[ 0,  1,  2],
    ...               [ 3,  4,  5],
    ...               [ 6,  7,  8],
    ...               [ 9, 10, 11]])
```

File: doc/source/reference/arrays.indexing.rst [hunk 2]
```diff
@@ -392,7 +396,7 @@ smaller than *x* it is identical to filling it with :const:`False`.
 
     >>> x = np.array([[1., 2.], [np.nan, 3.], [np.nan, np.nan]])
     >>> x[~np.isnan(x)]
-    array([ 1.,  2.,  3.])
+    array([1.,  2.,  3.])
```
Comment (eric-wieser) [inline doc/source/reference/arrays.indexing.rst]:
```suggestion
    array([1., 2., 3.])
```

File: doc/source/reference/arrays.indexing.rst [hunk 3]
```diff
@@ -447,7 +439,7 @@ also supports boolean arrays and will work without any surprises.
     advanced integer index. Using the :func:`ix_` function this can be done
     with:
 
-    >>> x = array([[ 0,  1,  2],
+    >>> x = np.array([[ 0,  1,  2],
     ...            [ 3,  4,  5],
     ...            [ 6,  7,  8],
     ...            [ 9, 10, 11]])
```
Comment (eric-wieser) [inline doc/source/reference/arrays.indexing.rst]:
```suggestion
    >>> x = np.array([[ 0,  1,  2],
    ...               [ 3,  4,  5],
    ...               [ 6,  7,  8],
    ...               [ 9, 10, 11]])
```

**Review by eric-wieser — APPROVED** [review]
Looks good, thanks.

Whoever merges should remember to squash

File: doc/source/reference/arrays.indexing.rst [hunk 4]
```diff
@@ -1,5 +1,9 @@
 .. _arrays.indexing:
 
+.. for doctests
+   >>> import numpy as np
```
Comment (seberg) [inline doc/source/reference/arrays.indexing.rst]: CircleCI is failing due to the `.. arrays.indexing:` being in the wrong spot now. My guess is that the for doctest must be below the Indexing title? (or above the `.. arrays.indexing:`).

Comment (mattip) [reply doc/source/reference/arrays.indexing.rst]: I canceled the CI runs that had not completed. @pdebuyl please run the refguide-check --rst test locally before pushing.

Comment (pdebuyl) [reply doc/source/reference/arrays.indexing.rst]: @mattip I did run refguide_check. Here, it is the sphinx build that errors, refguide_check did not complain.

Comment (mattip) [reply doc/source/reference/arrays.indexing.rst]: oops, sorry. I should have checked more carefully.

Comment (pdebuyl) [reply doc/source/reference/arrays.indexing.rst]: no worry :-)

**Comment by mattip** [comment]
Thanks @pdebuyl

**Comment by pdebuyl** [comment]
:-)

**Pull request merged**